gossipd/gossipd.c

```diff
@@ -455,10 +456,19 @@ static void dump_our_gossip(struct daemon *daemon, struct peer *peer)
 		/* Send channel_announce */
 		queue_peer_from_store(peer, &chan->bcast);
 
-		/* Send channel_update if we have one */
-		if (is_halfchan_defined(&chan->half[dir]))
-			queue_peer_from_store(peer, &chan->half[dir].bcast);
+		/* Send both channel_updates (if they exist): both help people
+		 * use our channel, so we care! */
+		for (int dir = 0; dir < 2; dir++) {
+			if (is_halfchan_defined(&chan->half[dir]))
+				queue_peer_from_store(peer, &chan->half[dir].bcast);
+		}
+
+		announced_channels = true;
 	}
+
+	/* If we announced a channel, we should send our own node_announcement */
+	if (announced_channels && me->bcast.index)
```

Comment:
It might be better to leave out the `announced_channels` flag.  I can't think of a scenario where it results in a different outcome than relying on the existence of our `node_announcement`.  Maybe it's useful once there's a spliced channel in progress if we wanted to stop gossiping about it in the interim?

Reply:
Hmm, good thinking!  Any failure is weird, and it can't hurt.  I'll remove the flag...